### PR TITLE
Improve configurability of MultiheadAttention projected query/key scaling.

### PR DESCRIPTION
**Pull request by tgunter**
Scaling the multihead-attention queries/keys prior to computing the logits is now handled by sub-modules.

### CODE REVIEW

File: axlearn/common/attention.py [hunk 1]
```diff
@@ -1877,8 +1971,8 @@ def _compute_logits(self, q_proj: Tensor, k_proj: Tensor) -> Tensor:
             # In the original XL-Net code, it applies scale on AC + BD:
             #
             # https://github.com/zihangdai/xlnet/blob/bbaa3a6fa0b3a2ee694e8cf66167434f9eca9660/modeling.py#L148
-            scale = self.per_head_dim() ** -0.5
-            logits = logits * scale
+            # with child_context("apply_scale_factor_logits", module=self):
```
Comment (markblee) [inline axlearn/common/attention.py:1974]: Intended?

Comment (tgunter) [reply axlearn/common/attention.py]: Oops, fixed, thanks.

File: axlearn/common/attention.py [hunk 2]
```diff
@@ -1847,13 +1939,12 @@ def forward(
 
     def _compute_logits(self, q_proj: Tensor, k_proj: Tensor) -> Tensor:
         cfg = self.config
-        if cfg.per_dim_scale is not None:
-            # Applies a per dim scale on q_proj.
-            q_proj = self.per_dim_scale(q_proj)
+        with child_context("apply_per_dim_scale", module=self):
+            q_proj = self.scale_query.apply_per_dim_scale(q_proj)
```
Comment (ruomingp) [inline axlearn/common/attention.py:1943]: Should we call `scale_query.apply_norm` before `apply_per_dim_scale`?

**Review by ruomingp — APPROVED** [review]

**Review by markblee — APPROVED** [review]

File: axlearn/common/attention.py (hunk 2 above)
Comment (tgunter) [reply axlearn/common/attention.py:1947]: Originally I wasn't going to, but in the end have now added it + a comment to clarify that only the scale_factor is applied to either/or logits/proj.

**Review by markblee — APPROVED** [review]
